frontend/lib/parsing/parsing-queries.cpp

```diff
@@ -388,21 +388,26 @@ void countTokens(Context* context, UniqueString path, ParserStats* parseStats) {
   }
 }
 
+static const BuilderResult*
+builderResultPtr(Context* context, ID id, UniqueString& pathOut) {
+  const BuilderResult* ret = nullptr;
+  UniqueString parentSymbolPath;
+  bool found = context->filePathForId(id, pathOut, parentSymbolPath);
+  if (found) {
+    const BuilderResult& br = parseFileToBuilderResult(context, pathOut,
+                                                       parentSymbolPath);
+    ret = &br;
+  }
+  return ret;
+}
+
 const Location& locateId(Context* context, ID id) {
   QUERY_BEGIN(locateId, context, id);
-
   Location result;
 
-  // Ask the context for the filename from the ID
```

Comment:
You dropped this comment, but I'd move it into `builderResultPtr`.

Reply:
I kind of feel like what it's doing is self evident from the name of the call.